composable_kernel/include/tensor_operation/gridwise_2d_reduction_blockwise.hpp

```diff
@@ -0,0 +1,863 @@
+          typename InElementwiseOperation,
+          typename OutElementwiseOperation>
+__global__ void kernel_reduce_blockwise(const In2dDescType in2dDesc,
+                                        const Out1dDescType out1dDesc,
+                                        const InElementwiseOperation in_elementwise_op,
+                                        const OutElementwiseOperation acc_elementwise_op,
+                                        AccDataType alpha,
+                                        const InDataType* const __restrict__ p_src_global,
+                                        OutDataType beta,
+                                        OutDataType* const __restrict__ p_dst_global,
+                                        const int* const __restrict__ ws_indices_global,
+                                        int* const __restrict__ indices_global)
+{
+    if constexpr(!NeedIndices)
+    {
+        GridwiseReduction::Run(in2dDesc,
+                               out1dDesc,
+                               in_elementwise_op,
+                               acc_elementwise_op,
+                               alpha,
+                               p_src_global,
+                               beta,
+                               p_dst_global,
+                               ws_indices_global,
+                               indices_global);
+    }
+    else
+    {
+        GridwiseReduction::RunWithIndices(in2dDesc,
+                                          out1dDesc,
+                                          in_elementwise_op,
+                                          acc_elementwise_op,
+                                          alpha,
+                                          p_src_global,
+                                          beta,
+                                          p_dst_global,
+                                          ws_indices_global,
+                                          indices_global);
+    };
+};
+
+template <typename GridwiseReduction,
+          bool NeedIndices,
+          typename InDataType,
+          typename OutDataType,
+          typename AccDataType,
+          typename In2dDescType,
+          typename Out1dDescType,
+          typename InElementwiseOperation,
+          typename OutElementwiseOperation>
+__global__ void
+kernel_reduce_blockwise_second_call(const In2dDescType in2dDesc,
+                                    const Out1dDescType out1dDesc,
+                                    const InElementwiseOperation in_elementwise_op,
+                                    const OutElementwiseOperation acc_elementwise_op,
+                                    AccDataType alpha,
+                                    const InDataType* const __restrict__ p_src_global,
+                                    OutDataType beta,
+                                    OutDataType* const __restrict__ p_dst_global,
+                                    const int* const __restrict__ ws_indices_global,
```

Comment:
https://github.com/ROCmSoftwarePlatform/composable_kernel/wiki/Coding-Style

Should use ```int``` in GPU code, use ```index_t``` instead

Reply:
Agree!